tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -1023,9 +1067,217 @@ def fit(
             calibrate_decision_threshold=calibrate_decision_threshold,
             infer_limit=infer_limit,
         )
-        self.save()
+        if dynamic_stacking:
+            logger.info(
+                "Dynamic stacking is enabled. AutoGluon will try to determine whether the input data is affected by stacked overfitting "
+                "and enable or disable stacking as a consequence."
+            )
+            num_stack_levels, time_limit = self._dynamic_stacking(ag_fit_kwargs, ag_post_fix_kwargs, **ds_args)
+
+            if (time_limit is not None) and (time_limit <= 0):
+                raise ValueError(
+                    f"Not enough time left to train models for the full fit. Consider specifying a larger time_limit. Time remaining: {time_limit}s"
+                )
+
+            ag_fit_kwargs["num_stack_levels"] = num_stack_levels
+            ag_fit_kwargs["time_limit"] = time_limit
+
+        self._fit(ag_fit_kwargs, ag_post_fix_kwargs)
+
         return self
 
+    def _fit(self, ag_fit_kwargs, ag_post_fix_kwargs):
+        self.save(silent=True)  # Save predictor to disk to enable prediction and training after interrupt
+        self._learner.fit(**ag_fit_kwargs)
+        self._set_post_fit_vars()
+        self._post_fit(**ag_post_fix_kwargs)
+        self.save()
+
+    def _dynamic_stacking(
+        self,
+        ag_fit_kwargs,
+        ag_post_fix_kwargs,
+        use_holdout=None,
+        detection_time_frac=None,
+        holdout_frac=None,
+        n_folds=None,
+        n_repeats=None,
+        memory_safe_fits=None,
+        clean_up_fits=None,
+        custom_val_data=None,
+    ):
```

Comment:
add a full docstring with type hints

Reply:
the argument for this function are validated in `_validate_fit_extra_kwargs` and the docstring/documentation is in the docstring/documentation of `fit()`. 

Do we want to have this in multiple places?

Reply:
As the docstrting for the parameters is in the docstring of`fit()`, I only added a "smaller" docstring here to explain the method but not the full docstring. 

Should I add a docstrting for the parameters here too, i.e., at two places?

Reply:
As long as it explains where one can find the definitions, it is fine.